riscv_isac/coverage.py

```diff
@@ -661,14 +713,15 @@ def compute_per_line(instr, cgf, xlen, addr_pairs,  sig_addrs):
     if instr.instr_name in ["fmadd.s","fmsub.s","fnmadd.s","fnmsub.s","fmadd.d","fmsub.d","fnmadd.d","fnmsub.d"]:
         rs3_val = '0x' + (arch_state.f_rf[rs3]).lower()
 
-    if instr.instr_name in ['csrrwi']  or instr.instr_name in ['fsrmi']:
-        arch_state.fcsr = instr.zimm
+    if instr.instr_name in ['csrrwi']:
+        csr_regfile.csr_regs["fcsr"] = instr.zimm
 
     #Having the rm value initiated before checking the conditions against instrucion names
     rm = instr.rm
-    if instr.instr_name in ["fadd.s","fsub.s","fmul.s","fdiv.s","fsqrt.s","fmadd.s","fmsub.s","fnmadd.s","fnmsub.s","fmax.s","fmin.s","feq.s","flt.s","fle.s","fmv.x.w","fmv.w.x","fcvt.wu.s","fcvt.s.wu","fcvt.w.s","fcvt.s.w","fsgnj.s","fsgnjn.s","fsgnjx.s","fclass.s","fcvt.s.l", "fcvt.s.lu","fadd.d","fsub.d","fmul.d","fdiv.d","fsqrt.d","fmadd.d","fmsub.d","fnmadd.d","fnmsub.d","fmax.d","fmin.d","feq.d","flt.d","fle.d","fld","fcvt.wu.d","fcvt.d.wu","fcvt.w.d","fcvt.d.w","fsgnj.d","fsgnjn.d","fsgnjx.d","fclass.d","fcvt.d.s", "fcvt.s.d","fsd","fcvt.d.l","fcvt.d.lu","fcvt.d.s","fcvt.d.w","fcvt.d.wu","fcvt.l.d","fcvt.lu.d","fcvt.s.d","fcvt.w.d","fcvt.wu.d","fmv.d.x","fmv.x.d"]:
+    #Checking the Floating point extension to define the rm  value respectively from the respectie csr in csr_regfile
+    if ('F' or 'D') in cgf[cov_labels]['config'][0].replace("check ISA:=regex",""):
```

Comment:
This check can be modifed to `if rm is not None`. That is a better check than relying on the config strings.